Src/library/net/network/socket/network_socket.e

```diff
@@ -206,35 +238,73 @@ feature -- Status setting
 
 	set_timeout (n: INTEGER)
 			-- Set timeout to `n' seconds.
+		obsolete
+			"Use `set_timeout_ns (ns)` [2018-01-15]"
 		require
 			non_negative: n >= 0
 		do
-			if n > 0 then
-				timeout := n
-			else
-				timeout := default_timeout
-			end
+			set_timeout_ns (n.to_natural_64)
 		ensure
 			timeout_set: timeout = n or timeout = default_timeout
 		end
 
+	set_timeout_ns (a_timeout_nanoseconds: NATURAL_64)
+			-- Set timeout with to `a_timeout_nanoseconds` nanoseconds.
+			-- warning: the timeout granularity of the platform may not be nanoseconds, but micro or milliseconds.
+		require
+			positive_timeout: a_timeout_nanoseconds /= 0
+			is_valid_timeout_ns: is_valid_timeout_ns (a_timeout_nanoseconds)
+		do
+			if a_timeout_nanoseconds = 0 then
```

Comment:
This cannot happen because of the precondition.

Reply:
indeed, wrong precondition